src/cascadia/TerminalConnection/ConhostConnection.idl

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.Terminal.TerminalConnection
     [default_interface]
     runtimeclass ConhostConnection : ITerminalConnection
     {
-        ConhostConnection(String cmdline, String startingDirectory, UInt32 rows, UInt32 columns);
+        ConhostConnection(String cmdline, String startingDirectory, UInt32 rows, UInt32 columns, Guid guid);

```

Comment:
Would you mind adding GetGuid() to ITerminalConnection? I think that'll complete the round trip here :smile:

Reply:
I wonder if it should be specified during construction, or generated by each individual connection type and only provided via GetGuid... what do you think?

Reply:
Ah, it wouldn't be GetGuid, it would be `Guid guid { get; }` (property getter magic)

Reply:
The way I've left it is that if you pass in the default guid (all zeroes basically) it will assign one, otherwise it will use the one specified.